sources/engine/Stride.Physics/Simulation.cs

```diff
@@ -756,11 +795,39 @@ static Constraint CreateConstraintInternal(ConstraintTypes type, RigidbodyCompon
                 constraintBase.RigidBodyB = rigidBodyB;
                 rigidBodyB.LinkedConstraints.Add(constraintBase);
             }
+            constraintBase.RigidBodyA = rigidBodyA;
             rigidBodyA.LinkedConstraints.Add(constraintBase);
 
             return constraintBase;
         }
 
+        static HingeConstraint CreateHingeConstraintInternal(RigidbodyComponent rigidBodyA, RigidbodyComponent rigidBodyB, Vector3 pivotInA, Vector3 pivotInB, Vector3 axisInA, Vector3 axisInB, bool useReferenceFrameA = false)
+        {
+            if (rigidBodyB != null && rigidBodyB.Simulation != rigidBodyA.Simulation) throw new Exception("Both RigidBodies must be on the same simulation");
```

Comment:
@xen2 what do you think about this specific coding style, I know it's fairly common for the rest of the code base to use such a style for exception throwing but not sure if we should avoid it since it doesn't match with the base style for ``if`` statement, should I comment on it when doing reviews ?

Reply:
I prefer two lines, but not so picky about one line version if it's short.
This one is quite long so probably better as 2 lines?